### DISCUSSION TITLE
NAV WP - Waypoint mode does not hit waypoints.

### DISCUSSION THREAD

**Discussion by Ktr128** [iNavFlight/inav#8627] [Q&A]
Wanted to do a low flight mission over the runway. But the plane flies past WP
![7](https://user-images.githubusercontent.com/98200485/206971026-0f1a88c7-4b04-4fbb-a191-cceaf9fcdecd.png)

Red line shows actual flight.
Auto Level, autotune, servo auto trim I did.
3D fix is ​​present. Sees more than 20 satellites.

Inav 5.1
FC Matek-H743 wing v3
GPS MATEK M10-5883
The compass is disabled.

Waypoint radius tried to reduce to 1000.
The only thing I see is to try to change the GPS Navigation Position XY parameter.

Black box disabled for this reason #8554
Please help, what needs to be done?

**Comment by Ktr128** [comment]
Diff all:

# start the command batch
batch start

# reset configuration to default settings
defaults noreboot

# resources

# mixer

mmix reset

mmix 0  1.000  0.000  0.000  0.000

# servo mixer
smix reset

smix 0 1 0 50 0 -1
smix 1 1 1 50 0 -1
smix 2 2 0 -50 0 -1
smix 3 2 1 50 0 -1

# servo
servo 1 1238 1938 1588 -100
servo 2 1237 1937 1587 100

# safehome

# feature
feature -BLACKBOX
feature -OSD
feature MOTOR_STOP
feature GPS
feature PWM_OUTPUT_ENABLE
feature FW_LAUNCH

# beeper

# blackbox
blackbox -NAV_ACC
blackbox NAV_POS
blackbox NAV_PID
blackbox MAG
blackbox ACC
blackbox ATTI
blackbox RC_DATA
blackbox RC_COMMAND
blackbox MOTORS

# map

# serial
serial 1 2 115200 115200 0 115200

# led

# color

# mode_color

# aux
aux 0 0 0 1725 2100
aux 1 1 1 1300 1700
aux 2 12 1 1750 2100
aux 3 10 2 1800 2100
aux 4 28 4 1800 2100
aux 5 37 6 1800 2100
aux 6 54 5 1800 2100

# adjrange

# rxrange

# temp_sensor

# wp
#wp 0 invalid

# osd_layout

# logic
logic 0 1 -1 1 2 18 0 1 0
logic 1 1 0 38 0 4 0 1750 0
logic 2 1 -1 12 2 17 0 0 0
logic 3 1 -1 18 0 2 0 0 0
logic 4 1 2 18 0 2 0 2 0
logic 5 1 0 18 0 2 0 1 0
logic 6 1 -1 1 5 2 0 0 0
logic 7 1 6 2 1 8 0 1750 0
logic 8 1 7 38 0 9 0 1950 0

# global vars

# programmable pid controllers

# master
set gyro_main_lpf_hz = 25
set dynamic_gyro_notch_q = 250
set dynamic_gyro_notch_min_hz = 30
set gyro_zero_x = 3
set gyro_zero_y = 9
set gyro_zero_z = 12
set ins_gravity_cmss =  1163.263
set acc_hardware = ICM42605
set acczero_x = -4
set acczero_y = 7
set acczero_z = -15
set accgain_x = 3547
set accgain_y = 4939
set accgain_z = 4926
set align_mag = CW270FLIP
set mag_hardware = NONE
set baro_hardware = SPL06
set serialrx_provider = CRSF
set motor_pwm_protocol = STANDARD
set failsafe_procedure = RTH
set align_board_roll = 2
set align_board_yaw = 2700
set current_meter_scale = 150
set platform_type = AIRPLANE
set model_preview_type = 8
set imu_dcm_ki = 0
set small_angle = 180
set imu_acc_ignore_rate = 9
set imu_acc_ignore_slope = 5
set applied_defaults = 3
set gps_sbas_mode = AUTO
set gps_ublox_use_galileo = ON
set airmode_type = STICK_CENTER_ONCE
set nav_wp_radius = 5000
set nav_wp_safe_distance = 0
set nav_rth_allow_landing = NEVER
set nav_rth_alt_mode = AT_LEAST_LINEAR_DESCENT
set nav_rth_abort_threshold = 0
set nav_rth_altitude = 10000
set nav_fw_bank_angle = 45
set nav_fw_dive_angle = 45
set nav_fw_loiter_radius = 10000
set nav_fw_control_smoothness = 2
set nav_fw_launch_accel = 1000
set nav_fw_launch_motor_delay = 0
set nav_fw_launch_timeout = 6000
set nav_fw_launch_max_altitude = 9000
set nav_fw_launch_climb_angle = 20

# profile
profile 1

set fw_p_pitch = 15
set fw_i_pitch = 5
set fw_d_pitch = 5
set fw_ff_pitch = 179
set fw_p_roll = 15
set fw_i_roll = 3
set fw_d_roll = 7
set fw_ff_roll = 117
set fw_p_yaw = 20
set fw_i_yaw = 0
set fw_ff_yaw = 100
set max_angle_inclination_rll = 550
set dterm_lpf_hz = 10
set fw_turn_assist_pitch_gain =  0.300
set nav_fw_pos_z_p = 35
set nav_fw_pos_xy_p = 70
set d_boost_min =  1.000
set d_boost_max =  1.000
set fw_level_pitch_trim = -0.486
set rc_expo = 30
set rc_yaw_expo = 30
set roll_rate = 11
set pitch_rate = 7
set yaw_rate = 3

# profile
profile 2


# profile
profile 3


# battery_profile
battery_profile 1

set vbat_min_cell_voltage = 270
set vbat_warning_cell_voltage = 280
set throttle_idle =  5.000
set nav_fw_cruise_thr = 1500
set nav_fw_min_thr = 1350
set nav_fw_max_thr = 1900
set nav_fw_launch_thr = 2000

# battery_profile
battery_profile 2


# battery_profile
battery_profile 3


# restore original profile selection
profile 1
battery_profile 1

# save configuration
save

#

**Comment by MrD-RC** [comment]
If this is on 5.1. Wait for 6.0 to come out. There are big improvements on the waypoint tracking, and flight in general.

**Reply by Ktr128** [reply]
Thank you for your advice!
When is INAV 6 going to be released?

**Reply by MrD-RC** [reply]
There is a feature preview pre-release available now. But I don't believe that has all the updates to the waypoint tracking. Release Candidates and stable will be around March 2023.

**Reply by b14ckyy** [reply]
Waypoint tracking is in there as an early version and not very precise yet. Also hard to properly tune. 
@Ktr128 do you have a blackbox log we can check, just to make sure it current behavior is indeed normal for 5.1?

**Reply by Ktr128** [reply]
Black box disabled for this reason #8554

**Comment by Ktr128** [comment]
Tell me please. If using an older version of inav, NAV WP will be accurate or is it the same as in inav 5.1 ?

**Reply by MrD-RC** [reply]
Older than 5.1 will not be better. 6.0 will be, when it is released.
